dev-packages/e2e-tests/test-applications/astro-5/tests/tracing.dynamic.test.ts

```diff
@@ -119,3 +119,147 @@ test.describe('tracing in dynamically rendered (ssr) routes', () => {
+    });
+
+    const serverHTTPServerRequestTxnPromise = waitForTransaction('astro-5', txnEvent => {
+      return txnEvent?.transaction?.startsWith('GET /api/user/') ?? false;
+    });
+
+    await page.goto('/user-page/myUsername123');
+
+    const clientPageloadTxn = await clientPageloadTxnPromise;
+    const serverPageRequestTxn = await serverPageRequestTxnPromise;
+    const serverHTTPServerRequestTxn = await serverHTTPServerRequestTxnPromise;
+    const serverRequestHTTPClientSpan = serverPageRequestTxn.spans?.find(
+      span => span.op === 'http.client' && span.description?.includes('/api/user/'),
+    );
+
+    const clientPageloadTraceId = clientPageloadTxn.contexts?.trace?.trace_id;
+
+    // Verify all spans have the same trace ID
+    expect(clientPageloadTraceId).toEqual(serverPageRequestTxn.contexts?.trace?.trace_id);
+    expect(clientPageloadTraceId).toEqual(serverHTTPServerRequestTxn.contexts?.trace?.trace_id);
+    expect(clientPageloadTraceId).toEqual(serverRequestHTTPClientSpan?.trace_id);
+
+    // serverPageRequest has no parent (root span)
+    expect(serverPageRequestTxn.contexts?.trace?.parent_span_id).toBeUndefined();
+
+    // clientPageload's parent and serverRequestHTTPClient's parent is serverPageRequest
+    const serverPageRequestSpanId = serverPageRequestTxn.contexts?.trace?.span_id;
+    expect(clientPageloadTxn.contexts?.trace?.parent_span_id).toEqual(serverPageRequestSpanId);
+    expect(serverRequestHTTPClientSpan?.parent_span_id).toEqual(serverPageRequestSpanId);
+
+    // serverHTTPServerRequest's parent is serverRequestHTTPClient
+    expect(serverHTTPServerRequestTxn.contexts?.trace?.parent_span_id).toEqual(serverRequestHTTPClientSpan?.span_id);
+  });
+
+  test('sends parametrized pageload, server and API request transaction names', async ({ page }) => {
+    const clientPageloadTxnPromise = waitForTransaction('astro-5', txnEvent => {
+      return txnEvent?.transaction?.startsWith('/user-page/') ?? false;
+    });
+
+    const serverPageRequestTxnPromise = waitForTransaction('astro-5', txnEvent => {
+      return txnEvent?.transaction?.startsWith('GET /user-page/') ?? false;
+    });
+
+    const serverHTTPServerRequestTxnPromise = waitForTransaction('astro-5', txnEvent => {
+      return txnEvent?.transaction?.startsWith('GET /api/user/') ?? false;
+    });
+
+    await page.goto('/user-page/myUsername123');
+
+    const clientPageloadTxn = await clientPageloadTxnPromise;
+    const serverPageRequestTxn = await serverPageRequestTxnPromise;
+    const serverHTTPServerRequestTxn = await serverHTTPServerRequestTxnPromise;
+
+    const serverRequestHTTPClientSpan = serverPageRequestTxn.spans?.find(
+      span => span.op === 'http.client' && span.description?.includes('/api/user/'),
+    );
+
+    // Client pageload transaction - actual URL with pageload operation
+    expect(clientPageloadTxn).toMatchObject({
+      transaction: '/user-page/myUsername123', // todo: parametrize to '/user-page/[userId]'
```

Comment:
not sure if you already thought of a way how to achieve this (if so, ignore me :D) but: We could inject the parameterized route we identify on the server into a <meta> tag just like we do with the trace data.

Reply:
I already found something :D
working on this in another PR. The parametrized route is in the context of the astro request so this is fairly easy.